src/platform/graphics/egl_extensions.cpp

```diff
@@ -69,7 +58,6 @@ mg::EGLExtensions::EGLExtensions() :
      */
     glEGLImageTargetTexture2DOES{
```

Comment:
This (and the `egl{Create,Destroy}ImageKHR` extensions above) also should really be checked properly. Only EGL client extensions (of which `EGL_EXT_platform_base` is the only one we use here) can be checked without already having an EGL Display.

That would make for a larger change, but also might make the rest of the changes a bit nicer - by making the big change you could pass in fully-functional extension objects, rather than having the two-step initialisation `optional<foo>` dance?

Reply:
I meant to ask about that. https://www.khronos.org/registry/EGL/extensions/KHR/EGL_KHR_image_base.txt doesn't mention if it's a client or display extension. Which string do we check?

Reply:
Everything that's not explicitly a client extension is a display extension. Client extensions didn't exist at all until [EGL_EXT_client_extensions](https://www.khronos.org/registry/EGL/extensions/EXT/EGL_EXT_client_extensions.txt), and that was a fair way into EGL 1.4's lifetime.

Reply:
Is `GL_OES_EGL_image_external` the correct name to test if `glEGLImageTargetTexture2DOES` is available? Or is that also `EGL_KHR_image_base`?

Reply:
You're looking for [GL_OES_EGL_image](https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image.txt), yes. Note that this is actually a GL extension, not an EGL extension, so it's not entirely clear to me why it's in `EGLExtensions`.